machine/v1beta1/types_azureprovider.go

```diff
@@ -174,6 +174,13 @@ type Image struct {
 	Version string `json:"version"`
 	// ResourceID specifies an image to use by ID
 	ResourceID string `json:"resourceID"`
+	// Type identifies the source of the image and related information, such as purchase plans.
+	// Valid values are "ID", "MarketplaceWithPlan", "MarketplaceNoPlan", and omitted, which
```

Comment:
I suggest reserving the right to change the meaning of `""` in this documentation.  Maybe something like 

Valid values are "ID"< "MarketplaceWithPlan", "MarketplaceNoPlan", and "".
"" means no opinion and the platform chooses a good default which may change over time. Currently that default is "MarketplaceNoPlan" if publisher data is supplied, or "ID" if not.
For more information***